PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,423 @@
+                $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) size has exceeded Organization DefaultPublicFolderProhibitPostQuota value!"
+                WritetoScreenANDlog -Issue $Issue -Fix $Fix
+            } else {
+                $Fix= "FIX --> Please follow the following article https://aka.ms/Setorgconfig to modify the Organization DefaultPublicFolderProhibitPostQuota:$($PublicFolderInfo.DefaultPublicFolderProhibitPostQuota) Bytes & DefaultPublicFolderIssueWarningQuota:$($PublicFolderInfo.DefaultPublicFolderIssueWarningQuota) Bytes values to accomodate the public folder $($PublicFolderInfo.Publicfolder.Identity) size:$($PublicFolderInfo.Publicfoldersize) Bytes"
+                $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) size has exceeded Organization DefaultPublicFolderProhibitPostQuota value!"
+                WritetoScreenANDlog -Issue $Issue -Fix $Fix
+            }
+        }
+    } else {
+        [Int64]$PFProhibitPostQuota=[Int64]$PublicFolderInfo.Publicfolder.ProhibitPostQuota.split("(")[1].split(" ")[0].replace(",", "")
+        if ($PublicFolderInfo.Publicfoldersize -ge $PFProhibitPostQuota) {
+            #validate if Giant PF to lower the PF size down else increase PFProhibitPostQuota or inhertit Org DefaultPublicFolderProhibitPostQuota
+            if ($PFProhibitPostQuota -ge 21474836480) {
+                $Fix= "FIX --> Please follow the following article https://aka.ms/Setorgconfig to modify the Organization DefaultPublicFolderProhibitPostQuota:$($PublicFolderInfo.DefaultPublicFolderProhibitPostQuota) Bytes & DefaultPublicFolderIssueWarningQuota:$($PublicFolderInfo.DefaultPublicFolderIssueWarningQuota) Bytes values to avoid having Giant public folders with sizes exceeding 20GB, then delete or move items from that giant folder $($PublicFolderInfo.Publicfolder.Identity) to reduce the size of that public folder to an appropriate size accomadating with newly modified organization values."
+                $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) size has exceeded Individual Public Folder ProhibitPostQuota value"
+                WritetoScreenANDlog -Issue $Issue -Fix $Fix
+            } else {
+                $Fix= "FIX --> Please follow the following article https://aka.ms/setpf to modify the Public Folder ProhibitPostQuota:$PFProhibitPostQuota Bytes value to accomodate the public folder $($PublicFolderInfo.Publicfolder.Identity) size:$($PublicFolderInfo.Publicfoldersize) Bytes"
+                $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) size has exceeded Individual Public Folder ProhibitPostQuota value"
+                WritetoScreenANDlog -Issue $Issue -Fix $Fix
+            }
+        }
+    }
+}
+Function ValidateDumpsterChildren {
+    param([PSCustomObject]$PublicFolderInfo)
+    $Haschildren= Get-PublicFolder $PublicFolderInfo.Publicfolderdumpster.EntryId -ErrorAction stop -GetChildren
```

Comment:
By having the `ErrorAction` set to `stop` here without being in a `try` `catch` would result in the script stopping here. Is that the intended result?

Reply:
Fixed by adding Try Catch as advised.